packages/playwright-test/src/matchers/expect.ts

```diff
@@ -106,27 +107,45 @@ export const printReceivedStringContainExpectedResult = (
 
 // #endregion
 
-type ExpectMessageOrOptions = undefined | string | { message?: string, timeout?: number, intervals?: number[] };
+type ExpectMessage = string | { message?: string };
 
-function createExpect(actual: unknown, messageOrOptions: ExpectMessageOrOptions, isSoft: boolean, isPoll: boolean, generator?: Generator): any {
-  return new Proxy(expectLibrary(actual), new ExpectMetaInfoProxyHandler(messageOrOptions, isSoft, isPoll, generator));
+function createExpect(actual: unknown, info: ExpectMetaInfo): any {
+  return new Proxy(expectLibrary(actual), new ExpectMetaInfoProxyHandler(info));
 }
 
 export const expect: Expect = new Proxy(expectLibrary, {
-  apply: function(target: any, thisArg: any, argumentsList: [actual: unknown, messageOrOptions: ExpectMessageOrOptions]) {
+  apply: function(target: any, thisArg: any, argumentsList: [unknown, ExpectMessage?]) {
     const [actual, messageOrOptions] = argumentsList;
-    return createExpect(actual, messageOrOptions, false /* isSoft */, false /* isPoll */);
+    const message = isString(messageOrOptions) ? messageOrOptions : messageOrOptions?.message;
+    return createExpect(actual, { message });
   }
 });
 
-expect.soft = (actual: unknown, messageOrOptions: ExpectMessageOrOptions) => {
-  return createExpect(actual, messageOrOptions, true /* isSoft */, false /* isPoll */);
+expect.soft = (actual: unknown, messageOrOptions?: ExpectMessage) => {
+  const message = isString(messageOrOptions) ? messageOrOptions : messageOrOptions?.message;
+  return createExpect(actual, { message, isSoft: true });
 };
 
-expect.poll = (actual: unknown, messageOrOptions: ExpectMessageOrOptions) => {
+expect.poll = (actual: unknown, messageOrOptions?: ExpectMessage & { timeout?: number, intervals?: number[] }) => {
   if (typeof actual !== 'function')
```

Comment:
I guess this check should now move somewhere else, because you can have `expect.configure({ poll: true })` that would completely avoid this?